Automattic/wp-calypso
# WooCommerce: Update products UI state to be keyed by siteId.

This PR updates the products (and variations, since they are located under products) UI state to be keyed by siteId. It also drops `payload` from the actions like we have done elsewhere. 

I wanted to do this before adding anything new to the state tree here (like tracking the current page of products for product listing, like #15042) so that we are storing the correct product information under the correct site. I also wanted to bring this code closer to what we are doing elsewhere (all of other state branches under `state.extensions.woocommerce.ui` are keyed by siteId). 

To test:
* Run `npm test` and make sure all tests pass.
* Go to `http://calypso.localhost:3000/store/product/:site`
* Test various parts of the form to make sure you can type inside input boxes, create variations, etc.

cc @kellychoffman Any recommendations on what to show if someone goes directly to the product form and we don't have a siteId loaded in yet? Right now I am just disabling the save button and hiding the form -- the sidebar already shows placeholders. I looked at what Calypso does for posting and they show the form -- but anything you type gets lost once the siteId loads in, which isn't great. Whatever we want to do, I will do in a smaller follow-up PR.
